File: frontend/lib/stats.ts
import { apiFetch } from "@/lib/api";

export type DailyActivity = {
  date: string;
  added: number;
  sent: number;
};

export type StatsOverview = {
  total_prospects: number;
  completed_count: number;
  generating_count: number;
  failed_count: number;
  sent_count: number;
  csv_upload_count: number;
  seller_knowledge_status: "NOT_GENERATED" | "GENERATING" | "COMPLETED" | "FAILED";
  has_seller_knowledge: boolean;
  daily_activity: DailyActivity[];
};

export function fetchStatsOverview(): Promise<StatsOverview> {
  return apiFetch<StatsOverview>("/api/stats/overview");
}

File: backend/app/schemas/stats.py
from pydantic import BaseModel

from app.models.seller_knowledge import SellerKnowledgeStatus


class DailyActivity(BaseModel):
    date: str
    added: int
    sent: int


class StatsOverview(BaseModel):
    total_prospects: int
    completed_count: int
    generating_count: int
    failed_count: int
    sent_count: int
    csv_upload_count: int
    seller_knowledge_status: SellerKnowledgeStatus
    has_seller_knowledge: bool
    daily_activity: list[DailyActivity]
